File: LeetCode_Solutions/LeetCode_Solutions/ExcelSheetColumnNumber_171.m
//
//  ExcelSheetColumnNumber_171.m
//  LeetCode_Solutions
//
//  Created by 来自遥远星系的核心巡洋舰 on 16/10/14.
//  Copyright © 2016年 KnightJoker. All rights reserved.
//

#import "ExcelSheetColumnNumber_171.h"

@implementation ExcelSheetColumnNumber_171

- (NSInteger)titleToNumber:(NSString *)s {

    if (s == nil) {
        return 0;
    }
    NSInteger sum = 0;
    for (int i = 0; i < s.length; i++) {
        sum = 26 * sum + [s characterAtIndex:i] - 'A' + 1;
    }
    return sum;
}

@end

File: LeetCode_Swift/LeetCode_Swift/ExcelSheetColumnNumber_171.swift
//
//  ExcelSheetColumnNumber_171.swift
//  LeetCode_Swift
//
//  Created by Huni on 11/08/2017.
//  Copyright © 2017 KnighhtJoker. All rights reserved.
//

import Foundation

public class ExcelSheetColumnNumber_171 {

    func titleToNumber(_ s: String) -> Int {
        
        if s.characters.count == 0 {
            return 0
        }
        var sum = 0
        for char in s.characters {
            sum = 26 * sum + Int(String(char).unicodeScalars[String(char).unicodeScalars.startIndex].value - "A".unicodeScalars["A".unicodeScalars.startIndex].value) + 1;
        }
        return sum
    }
    
}
